src/run_perf_test.py

```diff
@@ -45,14 +49,22 @@ def main():
     manifest = BundleManifest.from_file(args.bundle_manifest)
     config = yaml.safe_load(args.config)
 
+    security = "security" in manifest.components and not args.insecure
+    if security:
+        tests_dir = os.path.join(os.getcwd(), "test-results", "perf-test", "with-security")
+    else:
+        tests_dir = os.path.join(os.getcwd(), "test-results", "perf-test", "without-security")
+    os.makedirs(tests_dir, exist_ok=True)
+
     with TemporaryDirectory(keep=args.keep, chdir=True) as work_dir:
         current_workspace = os.path.join(work_dir.name, "infra")
         with GitRepository(get_infra_repo_url(), "main", current_workspace):
-            security = "security" in manifest.components
             with WorkingDirectory(current_workspace):
-                with PerfTestCluster.create(manifest, config, args.stack, security, current_workspace) as (test_cluster_endpoint, test_cluster_port):
-                    perf_test_suite = PerfTestSuite(manifest, test_cluster_endpoint, security, current_workspace, args)
-                    perf_test_suite.execute()
+                with PerfTestCluster.create(manifest, config, args.stack, security, current_workspace) \
+                        as (test_cluster_endpoint, test_cluster_port):
+                    perf_test_suite = PerfTestSuite(manifest, test_cluster_endpoint, security,
+                                                    current_workspace, tests_dir, args)
+                    retry_call(perf_test_suite.execute, tries=3, delay=60, backoff=2)
```

Comment:
Should we expose retry options to the command line?

Reply:
That seems like a good option to have. I do have a few things to follow up on post based on the test run outcomes once everything is up and running. 
Added an issue here for tracking - https://github.com/opensearch-project/OpenSearch/issues/2718